react-app/src/components/SaveOutputForm.js

```diff
@@ -0,0 +1,213 @@
+import React, { useState } from 'react';
+import { Alert, Button, Col, Form, Row } from 'react-bootstrap';
+import fhirpath from 'fhirpath';
+
+function getLabel(bundle, id) {
+  // attempt to get MRN and name with fhirpath
+  const patient = fhirpath.evaluate(bundle, "Bundle.descendants().resource.where(resourceType='Patient')")[0];
+  if (patient) {
+    const mrn = patient.id;
+    const name = patient.name[0].text;
+
+    const isMasked = fhirpath.evaluate(patient, 'Patient.identifier.extension.valueCode')[0] === 'masked';
+
+    // if both MRN and name -- return string w / both
+    if (typeof mrn === 'string' && mrn.length > 0 && typeof name === 'string' && name.length > 0 && !isMasked) {
+      const label = mrn.concat(': ').concat(name);
+      return label;
+    }
+    // if either MRN or name -- return string / the available one
+    if (typeof name === 'string' && name.length > 0) {
+      return name;
+    }
+    // if neither MRN nor name -- "Patient " + patient_resource_id
+    if (typeof mrn === 'string' && mrn.length > 0) {
+      return mrn;
+    }
+  }
+
+  // if no patient resource ID -- return "Patient " + props.id
+  let label = 'Patient';
+  label = label.concat(' ').concat(id.toString());
+  return label;
+}
+
+function SaveOutputForm(props) {
+  const [selectAll, setSelectAll] = useState(true);
+  const [saveLogs, setSaveLogs] = useState(false);
+
+  const [showSavedAlert, setShowSavedAlert] = useState(false);
+  const [showErrorAlert, setShowErrorAlert] = useState(false);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [showNoFilesAlert, setShowNoFilesAlert] = useState(false);
+
+  let defaultWhichFiles = {};
+  props.extractedData.forEach((bundle, i) => {
+    const label = getLabel(bundle, i);
+    defaultWhichFiles = {
+      ...defaultWhichFiles,
+      [label]: true,
+    };
+  });
```

Comment:
I don't think we need full object destructuring here, right? We just need to add a new key-value pair onto the existing object.

Reply:
I have modified to get rid of the object destructuring.